File: dotnet-adapters-shortest-paths-core/main/java/com/programmerare/shortestpaths/utils/TimeMeasurer.java
/*
* Copyright (c) Tomas Johansson , http://www.programmerare.com
* The code is made available under the terms of the MIT License.
* https://github.com/TomasJohansson/adapters-shortest-paths/blob/master/adapters-shortest-paths-core/License.txt
*/
package com.programmerare.shortestpaths.utils;

import java.util.Date;

public final class TimeMeasurer {

	public static TimeMeasurer start() {
		return new TimeMeasurer();
	}
	
	private final long startTime;
	
	private TimeMeasurer() {
		this.startTime = (new Date()).getTime();
	}
	
	public long getSeconds() {
		return ((new Date()).getTime()-startTime) / 1000;
	}
}

File: dotnet-adapters-shortest-paths-core/src/com.programmerare.shortestpaths/utils/TimeMeasurer.cs
/*
* Copyright (c) Tomas Johansson , http://www.programmerare.com
* The code is made available under the terms of the MIT License.
* https://github.com/TomasJohansson/adapters-shortest-paths/blob/master/adapters-shortest-paths-core/License.txt
*/
namespace com.programmerare.shortestpaths.utils
{
    public sealed class TimeMeasurer {

	    public static TimeMeasurer Start() {
		    return new TimeMeasurer();
	    }
	
	    private readonly long startTime;
	
	    private TimeMeasurer() {
		    this.startTime = 0;// TODO(new Date()).getTime();
	    }
	
	    public long GetSeconds() {
		    return 0;// TODO((new Date()).getTime()-startTime) / 1000;
	    }
    }
}